### PR TITLE
support forkserver in multiprocess executor

### LINKED ISSUE #6439: Poor multiprocess_executor performance on Mac
## Summary
I've tried launching a slightly modified example from [here](https://docs.dagster.io/concepts/ops-jobs-graphs/nesting-graphs) on both Mac and Linux (environment details below) using the multiprocess executor. Specifically, the code I'm running is below.

I'm running in two clean Python 3.9 virtual environments, populated only with ```pip install dagster dagit```. I also do not have ```$DAGSTER_HOME``` set, so there's no centralized configuration.

On Linux, this runs using the multiprocess executor in 8.84 seconds. **On Mac, it runs in ~2 minutes**. Additionally, there are many more event notifications on Mac (e.g., ```LOGS_CAPTURED```, ```STEP_START```, etc.).

Changing to ```in_process``` takes this down to ~1 second on both Mac and Linux. It's understandable that the multiprocessing overhead slows down jobs with fast ops, but I figured the 2 minute execution time on Mac was probably indicative of an issue.

## Reproduction
```python
from dagster import job, op

@op
def return_fifty():
    return 50.0

@op
def add_thirty_two(number):
    return number + 32.0

@op
def multiply_by_one_point_eight(number):
    return number * 1.8

@op
def log_number(context, number):
    context.log.info(f"number: {number}")

@job
def all_together_unnested():
    log_number(add_thirty_two(multiply_by_one_point_eight(return_fifty())))

if __name__ == "__main__":
    all_together_unnested.execute_in_process(
        run_config={
            "execution": {
                "config": {
                    "multiprocess": {
                        "max_concurrent": 4
                    }
                }
            }
        }
    )
```

Executed as:
```text
dagster pipeline execute -f test.py
```

Here is the log on Linux:

```text
Using temporary directory /SNIP/tmpa8bi1gze for storage. This will be removed when ``dagster pipeline execute`` exits.
To persist information across sessions, set the environment variable DAGSTER_HOME to a directory to use.

2022-01-31 20:18:11 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - RUN_START - Started execution of run for "all_together_unnested".
2022-01-31 20:18:11 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - ENGINE_EVENT - Executing steps using multiprocess executor: parent process (pid: 68105)
2022-01-31 20:18:11 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - return_fifty - ENGINE_EVENT - Launching subprocess for return_fifty
2022-01-31 20:18:13 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68172 - return_fifty - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 20:18:13 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68172 - return_fifty - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 20:18:14 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - multiply_by_one_point_eight - ENGINE_EVENT - Launching subprocess for multiply_by_one_point_eight
2022-01-31 20:18:15 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68197 - multiply_by_one_point_eight - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 20:18:15 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68197 - multiply_by_one_point_eight - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 20:18:16 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - add_thirty_two - ENGINE_EVENT - Launching subprocess for add_thirty_two
2022-01-31 20:18:17 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68218 - add_thirty_two - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 20:18:17 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68218 - add_thirty_two - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 20:18:18 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - log_number - ENGINE_EVENT - Launching subprocess for log_number
2022-01-31 20:18:19 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68242 - log_number - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 20:18:19 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68242 - log_number - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 20:18:20 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - ENGINE_EVENT - Multiprocess executor: parent process exiting after 8.84s (pid: 68105)
2022-01-31 20:18:20 -0500 - dagster - DEBUG - all_together_unnested - 99d3f74d-7f37-40d9-b8a5-ec083b6e417b - 68105 - RUN_SUCCESS - Finished execution of run for "all_together_unnested".
```

Here is the log on Mac:

```text
Using temporary directory /SNIP/tmpmhqtqr5k for storage. This will be removed when ``dagster pipeline execute`` exits.
To persist information across sessions, set the environment variable DAGSTER_HOME to a directory to use.

2022-01-31 17:19:32 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - RUN_START - Started execution of run for "all_together_unnested".
2022-01-31 17:19:32 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - ENGINE_EVENT - Executing steps using multiprocess executor: parent process (pid: 8520)
2022-01-31 17:19:32 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - return_fifty - ENGINE_EVENT - Launching subprocess for return_fifty
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - LOGS_CAPTURED - Started capturing logs for step: return_fifty.
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - STEP_START - Started execution of step "return_fifty".
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - STEP_OUTPUT - Yielded output "result" of type "Float". (Type check passed).
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - return_fifty - Writing file at: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/return_fifty/result
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - HANDLED_OUTPUT - Handled output "result" using IO manager "io_manager"
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8526 - return_fifty - STEP_SUCCESS - Finished execution of step "return_fifty" in 14ms.
2022-01-31 17:19:33 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - multiply_by_one_point_eight - ENGINE_EVENT - Launching subprocess for multiply_by_one_point_eight
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - LOGS_CAPTURED - Started capturing logs for step: multiply_by_one_point_eight.
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - STEP_START - Started execution of step "multiply_by_one_point_eight".
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - multiply_by_one_point_eight - Loading file from: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/return_fifty/result
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - LOADED_INPUT - Loaded input "number" using input manager "io_manager", from output "result" of step "return_fifty"
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - STEP_INPUT - Got input "number" of type "Any". (Type check passed).
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - STEP_OUTPUT - Yielded output "result" of type "Float". (Type check passed).
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - multiply_by_one_point_eight - Writing file at: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/multiply_by_one_point_eight/result
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - HANDLED_OUTPUT - Handled output "result" using IO manager "io_manager"
2022-01-31 17:20:13 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8533 - multiply_by_one_point_eight - STEP_SUCCESS - Finished execution of step "multiply_by_one_point_eight" in 15ms.
2022-01-31 17:20:14 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - add_thirty_two - ENGINE_EVENT - Launching subprocess for add_thirty_two
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - LOGS_CAPTURED - Started capturing logs for step: add_thirty_two.
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - STEP_START - Started execution of step "add_thirty_two".
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - add_thirty_two - Loading file from: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/multiply_by_one_point_eight/result
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - LOADED_INPUT - Loaded input "number" using input manager "io_manager", from output "result" of step "multiply_by_one_point_eight"
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - STEP_INPUT - Got input "number" of type "Any". (Type check passed).
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - STEP_OUTPUT - Yielded output "result" of type "Float". (Type check passed).
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - add_thirty_two - Writing file at: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/add_thirty_two/result
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - HANDLED_OUTPUT - Handled output "result" using IO manager "io_manager"
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8571 - add_thirty_two - STEP_SUCCESS - Finished execution of step "add_thirty_two" in 17ms.
2022-01-31 17:20:56 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - log_number - ENGINE_EVENT - Launching subprocess for log_number
2022-01-31 17:21:38 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - ENGINE_EVENT - Starting initialization of resources [io_manager].
2022-01-31 17:21:38 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - ENGINE_EVENT - Finished initialization of resources [io_manager].
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - LOGS_CAPTURED - Started capturing logs for step: log_number.
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - STEP_START - Started execution of step "log_number".
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - log_number - Loading file from: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/add_thirty_two/result
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - LOADED_INPUT - Loaded input "number" using input manager "io_manager", from output "result" of step "add_thirty_two"
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - STEP_INPUT - Got input "number" of type "Any". (Type check passed).
2022-01-31 17:21:39 -0800 - dagster - INFO - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - log_number - number: 122.0
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - STEP_OUTPUT - Yielded output "result" of type "Any". (Type check passed).
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - log_number - Writing file at: /Users/johnss51/Development/dagster/storage/5ea9c874-a495-44af-b855-021e6a496799/log_number/result
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - HANDLED_OUTPUT - Handled output "result" using IO manager "io_manager"
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8584 - log_number - STEP_SUCCESS - Finished execution of step "log_number" in 20ms.
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - ENGINE_EVENT - Multiprocess executor: parent process exiting after 2m6s (pid: 8520)
2022-01-31 17:21:39 -0800 - dagster - DEBUG - all_together_unnested - 5ea9c874-a495-44af-b855-021e6a496799 - 8520 - RUN_SUCCESS - Finished execution of run for "all_together_unnested".
```

## Additional Info about Your Environment

**Mac details:**
* MacOS 11.6.1
* Python 3.9.7 ([Clang 10.0.0 ] :: Anaconda, Inc. on darwin)
```text
alembic==1.6.5
aniso8601==7.0.0
attrs==21.4.0
bleach==4.1.0
certifi==2021.10.8
charset-normalizer==2.0.11
click==7.1.2
coloredlogs==14.0
croniter==1.2.0
dagit==0.13.17
dagster==0.13.17
dagster-graphql==0.13.17
defusedxml==0.7.1
docstring-parser==0.13
entrypoints==0.3
Flask==1.1.4
Flask-Cors==3.0.10
Flask-GraphQL==2.0.1
Flask-Sockets==0.2.1
gevent==21.12.0
gevent-websocket==0.10.1
gql==2.0.0
graphene==2.1.9
graphql-core==2.3.2
graphql-relay==2.0.1
graphql-server-core==1.2.0
graphql-ws==0.3.1
greenlet==1.1.2
grpcio==1.43.0
grpcio-health-checking==1.43.0
humanfriendly==10.0
idna==3.3
ipython-genutils==0.2.0
itsdangerous==1.1.0
Jinja2==2.11.3
jsonschema==4.4.0
jupyter-core==4.9.1
Mako==1.1.6
MarkupSafe==2.0.1
mistune==0.8.4
nbconvert==5.6.1
nbformat==5.1.3
packaging==21.3
pandocfilters==1.5.0
pendulum==2.1.2
promise==2.3
protobuf==3.19.4
Pygments==2.11.2
pyparsing==3.0.7
pyrsistent==0.18.1
python-dateutil==2.8.2
python-editor==1.0.4
pytz==2021.3
pytzdata==2020.1
PyYAML==6.0
requests==2.27.1
Rx==1.6.1
six==1.16.0
SQLAlchemy==1.4.31
tabulate==0.8.9
testpath==0.5.0
toposort==1.7
tqdm==4.62.3
traitlets==5.1.1
typing-compat==0.1.0
urllib3==1.26.8
watchdog==2.1.6
webencodings==0.5.1
Werkzeug==1.0.1
zope.event==4.5.0
zope.interface==5.4.0
```

**Linux details:**
* Red Hat 7.9
* Python 3.9.7 ([GCC 7.5.0] :: Anaconda, Inc. on linux)
```text
alembic==1.6.5
aniso8601==7.0.0
attrs==21.4.0
bleach==4.1.0
certifi==2021.10.8
charset-normalizer==2.0.11
click==7.1.2
coloredlogs==14.0
croniter==1.2.0
dagit==0.13.17
dagster==0.13.17
dagster-graphql==0.13.17
defusedxml==0.7.1
docstring-parser==0.13
entrypoints==0.3
Flask==1.1.4
Flask-Cors==3.0.10
Flask-GraphQL==2.0.1
Flask-Sockets==0.2.1
gevent==21.12.0
gevent-websocket==0.10.1
gql==2.0.0
graphene==2.1.9
graphql-core==2.3.2
graphql-relay==2.0.1
graphql-server-core==1.2.0
graphql-ws==0.3.1
greenlet==1.1.2
grpcio==1.43.0
grpcio-health-checking==1.43.0
humanfriendly==10.0
idna==3.3
ipython-genutils==0.2.0
itsdangerous==1.1.0
Jinja2==2.11.3
jsonschema==4.4.0
jupyter-core==4.9.1
Mako==1.1.6
MarkupSafe==2.0.1
mistune==0.8.4
nbconvert==5.6.1
nbformat==5.1.3
packaging==21.3
pandocfilters==1.5.0
pendulum==2.1.2
promise==2.3
protobuf==3.19.4
Pygments==2.11.2
pyparsing==3.0.7
pyrsistent==0.18.1
python-dateutil==2.8.2
python-editor==1.0.4
pytz==2021.3
pytzdata==2020.1
PyYAML==6.0
requests==2.27.1
Rx==1.6.1
six==1.16.0
SQLAlchemy==1.4.31
tabulate==0.8.9
testpath==0.5.0
toposort==1.7
tqdm==4.62.3
traitlets==5.1.1
typing-compat==0.1.0
urllib3==1.26.8
watchdog==2.1.6
webencodings==0.5.1
Werkzeug==1.0.1
zope.event==4.5.0
zope.interface==5.4.0
```

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by alangenfeld**
## Summary

Make the multiprocessing executor configurable in how it starts its subprocesses. Instead of only `spawn` we now also support `forkserver`.

We could potentially work to support `fork` but we use threads for cacnelation so out of the box I dont think it will work.

This is a little tricky since not all OS support `forkserver` so we have to be careful about that. 

That said, `forkserver` with the dagster preload is a big win for fast op Jobs so I think its worth persuing making that default when available. Will put that in a follow up diff.

resolves #6439

## Test Plan

added a basic test, can add more

going to fire a PR that changes the default to `forkserver` and see what happens in BK

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6446** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6446" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: python_modules/dagster/dagster/core/definitions/executor_definition.py [hunk 1]
```diff
@@ -255,21 +255,47 @@ def execute_in_process_executor(_):
...
+        Enum(
+            "StartMethod",
+            [
+                EnumValue(
+                    config_value="spawn",
+                    description="Start a fresh python interpreter process",
+                ),
+                EnumValue(
+                    config_value="forkserver",
+                    description="A server process is started and then subprocesses are forked from that.",
```
Comment (mgasner) [inline python_modules/dagster/dagster/core/definitions/executor_definition.py:280]: maybe link to the python docs for these

File: python_modules/dagster/dagster/core/executor/multiprocess.py [hunk 2]
```diff
@@ -85,10 +85,24 @@ def execute(self):
...
+
         self._retries = check.inst_param(retries, "retries", RetryMode)
         max_concurrent = max_concurrent if max_concurrent else multiprocessing.cpu_count()
-        self.max_concurrent = check.int_param(max_concurrent, "max_concurrent")
+        self._max_concurrent = check.int_param(max_concurrent, "max_concurrent")
+        start_method = check.str_param(start_method, "start_method")
+        valid_starts = multiprocessing.get_all_start_methods()
+        if start_method not in valid_starts:
+            raise DagsterUnmetExecutorRequirementsError(
+                f"The selected start_method '{start_method}' is not available. "
```
Comment (mgasner) [inline python_modules/dagster/dagster/core/executor/multiprocess.py:99]: maybe also print the os/python version (e.g. `sys.platform` and `sys.version`) to make our lives easier when helping users debug

**Comment by alangenfeld** [comment]
A comparison between forkserver and spawn for a simple job 

Screen Shot 2022-02-03 at 2 53 41 PM

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/snapshots/snap_test_pipeline_snapshot.py [hunk 3]
```diff
@@ -45,7 +45,7 @@
       }
     ],
     "name": "csv_hello_world",
-    "pipelineSnapshotId": "b9cba8fc3833bd35a92d085c833fb2c176a1c3b0",
+    "pipelineSnapshotId": "4a20d826ecc8a9a7728ec4bb15b6ab8e727e5f85",
```
Comment (gibsondan) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/snapshots/snap_test_pipeline_snapshot.py:48]: no back-compat issues being introduced here?

File: python_modules/dagster/dagster/core/executor/child_process_executor.py [hunk 4]
```diff
@@ -103,7 +103,7 @@ def _poll_for_event(process, event_queue):
     return None
 
 
-def execute_child_process_command(command):
+def execute_child_process_command(multiprocessing, command):
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/executor/child_process_executor.py:106]: this name might benefit from a noun of some sort - multiprocess_context?

**Review by gibsondan — CHANGES_REQUESTED** [review]
for the back-compat / snapshot question - I don't see a test change that would have caused it to change naturally and I think last time the default snapshot ID was changed it caused issues?

**Review by gibsondan — APPROVED** [review]
my confusion was that executors are user code, I am no longer confused

**Pull request merged**